testing/mock/test_system.cpp

```diff
@@ -276,11 +280,13 @@ void test_system::prepare_syfs(const test_platform &platform) {
     root_ = std::string(tmp);
     std::string cmd = "tar xzf " + std::string(platform.mock_sysfs) + " -C " +
                       root_ + " --strip 1";
-    std::system(cmd.c_str());
+    result = std::system(cmd.c_str());
   }
+  return (void) result;
```

Comment:
rather than doing this, can we throw an exception if the result is non-zero?
I put a sample of how to do it in an earlier review.